### DISCUSSION TITLE
useAsyncData to properly de-duplicated across requests

### DISCUSSION THREAD

**Discussion by hirotaka** [nuxt/nuxt#22960] [Questions] [answered]
Hi, I'm building an app using Rails on the back end and Nuxt3 on the front end.

I have a question about `useAsyncData`. According to the documentation,

https://nuxt.com/docs/api/composables/use-async-data#params

> key: a unique key to ensure that data fetching can be properly de-duplicated across requests. If you do not provide a key, then a key that is unique to the file name and line number of the instance of [useAsyncData](https://nuxt.com/docs/api/composables/use-async-data) will be generated for you.

I'm expecting the following code to generate only one request, but it actually generates two requests.

https://stackblitz.com/edit/nuxt-starter-m3bxac

![CleanShot 2023-09-02 at 20 09 10](https://github.com/nuxt/nuxt/assets/52546/c01f9cb2-9c12-4b9a-bc0e-e57ffcda5232)

Am I missing something or can you tell me a better way to do this?

**Comment by xjccc** [comment, marked as answer]
![image](https://github.com/nuxt/nuxt/assets/16478776/924389ac-b89a-4eb9-8c8b-8dd0bee620ff)

this is the explanation?

**Reply by hirotaka** [reply]
Thanks for the answer! I understood.
However, I have a new question, so I have created a new Discussion(https://github.com/nuxt/nuxt/discussions/23294)
